s3transfer/tasks.py

```diff
@@ -245,6 +157,97 @@ def _get_all_main_kwargs(self):
             kwargs[key] = result
         return kwargs
 
+    def _submit_task(self, executor, task):
+        future = executor.submit(task)
+        # Add this created future to the list of associated future just
+        # in case it is needed during cleanups.
+        self._transfer_coordinator.add_associated_future(future)
+        return future
+
+
+class SubmissionTask(Task):
+    """A base class for any submission task
+
+    Submission tasks are the top-level task used to submit a series of tasks
+    to execute a particular transfer.
+    """
+    def _main(self, transfer_future, **kwargs):
+        """
+        :type transfer_future: s3transfer.futures.TransferFuture
+        :param transfer_future: The transfer future associated with the
+            transfer request that tasks are being submitted for
+
+        :param kwargs: Any additional kwargs that you may want to pass
+            to the _submit() method
+        """
+        try:
+            # Before submitting any tasks, run all of the on_queued callbacks
+            on_queued_callbacks = get_callbacks(transfer_future, 'queued')
+            for on_queued_callback in on_queued_callbacks:
+                on_queued_callback()
+
+            # Once callbacks have been ran set the status to running.
+            self._transfer_coordinator.set_status_to_running()
```

Comment:
So I guess this depends on the definition of ``running``. Should it be when the first transfer API request is made like ``GetObject`` or ``PutObject`` or should it be once work has started for the request? I saw it as the later where if I called ``HeadObject`` or pulled data from the stream for an upload then my transfer has started and thus I put it in the "running" state. What do you think?